client/lib/query-manager/index.js

```diff
@@ -104,6 +104,27 @@ export default class QueryManager {
 	}
 
 	/**
+	 * A sorting function that defines the sort order of items under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Array}  keys  Keys to be sorted
+	 * @param  {Array}  items Items by which to sort
+	 * @param  {Object} query Query object
+	 * @return {Array}        Sorted keys
+	 */
+	sort( keys, items, query ) {
+		return keys.sort( ( keyA, keyB ) => {
+			if ( ! items[ keyA ] || ! items[ keyB ] ) {
```

Comment:
Hmm, is that really so much simpler? Looks like it's mostly just inversing the logic. I think I'd rather have the more exceptional case (element removed) treated first (by the conditional) and the `return` outside the conditional handle the more regular case...